librz/core/cmd_debug.c

```diff
@@ -1443,376 +1357,462 @@ static bool get_bin_info(RzCore *core, const char *file, ut64 baseaddr, PJ *pj,
+			const char *file = map->file ? map->file : map->name;
+			char *name = rz_str_escape((char *)rz_file_basename(file));
+			char *filesc = rz_str_escape(file);
+			/* TODO: do not spawn. use RzBin API */
+			if (sectname) {
+				char *sect = rz_str_escape(sectname);
+				res = rz_sys_cmd_strf("env RZ_BIN_PREFIX=\"%s\" rz-bin %s-B 0x%08" PFMT64x " -S \"%s\" | grep \"%s\"", name, mode, baddr, filesc, sect);
+				free(sect);
+			} else {
+				res = rz_sys_cmd_strf("env RZ_BIN_PREFIX=\"%s\" rz-bin %s-B 0x%08" PFMT64x " -S \"%s\"", name, mode, baddr, filesc);
+			}
+			free(filesc);
+			rz_cons_println(res);
+			free(name);
+			free(res);
+			if (libname || addr != UT64_MAX) { //only single match requested
+				break;
+			}
+		}
+	}
+	return RZ_CMD_STATUS_OK;
+}
+// dml
+RZ_IPI RzCmdStatus rz_cmd_debug_dml_handler(RzCore *core, int argc, const char **argv) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	RzListIter *iter;
+	RzDebugMap *map;
+	ut64 addr = core->offset;
+	rz_debug_map_sync(core->dbg); // update process memory maps
+	rz_list_foreach (core->dbg->maps, iter, map) {
+		if (addr >= map->addr && addr < map->addr_end) {
+			size_t sz;
+			char *buf = rz_file_slurp(argv[1], &sz);
+			//TODO: use mmap here. we need a portable implementation
+			if (!buf) {
+				eprintf("Cannot allocate 0x%08" PFMT64x " byte(s)\n", map->size);
+				return RZ_CMD_STATUS_ERROR;
+			}
+			rz_io_write_at(core->io, map->addr, (const ut8 *)buf, sz);
+			if (sz != map->size)
+				eprintf("File size differs from region size (%" PFMT64u " vs %" PFMT64d ")\n",
+					(ut64)sz, map->size);
+			eprintf("Loaded %" PFMT64u " byte(s) into the map region at 0x%08" PFMT64x "\n",
+				(ut64)sz, map->addr);
+			free(buf);
+			return RZ_CMD_STATUS_OK;
+		}
+	}
+	eprintf("No debug region found here\n");
+	return RZ_CMD_STATUS_ERROR;
+}
+
+// dmL
+RZ_IPI RzCmdStatus rz_cmd_debug_dmL_handler(RzCore *core, int argc, const char **argv) {
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	int size;
+	ut64 addr;
+	addr = core->offset;
+	size = (int)rz_num_math(core->num, argv[1]);
+	rz_debug_map_alloc(core->dbg, addr, size, true);
+	return RZ_CMD_STATUS_OK;
+}
+
+// dmw
+RZ_IPI int rz_cmd_debug_heap_windows(void *data, const char *input) {
+	RzCore *core = (RzCore *)data;
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
+#if __WINDOWS__
+	cmd_debug_map_heap_win(core, input);
+	return RZ_CMD_STATUS_OK;
+#else
+	eprintf("MALLOC algorithm not supported\n");
+	return RZ_CMD_STATUS_ERROR;
+#endif
+}
+
+// dmx
+RZ_IPI int rz_cmd_debug_heap_jemalloc(void *data, const char *input) {
+	RzCore *core = (RzCore *)data;
+	if (rz_debug_is_dead(core->dbg)) {
+		rz_cons_println("Debugging is not enabled. Run ood?");
+		return RZ_CMD_STATUS_ERROR;
+	}
```

Comment:
```suggestion
#define CMD_CHECK_DEBUG_DEAD(core) do { \
  if (rz_debug_is_dead(core->dbg)) { \
    rz_cons_println("Debugging is not enabled. Run ood?"); \
    return RZ_CMD_STATUS_ERROR; \
  }
} while(0)

	CMD_CHECK_DEBUD_DEAD(core);
```
better to avoid all the repetitions. This is probably the easy thing to reduce that.